tests/test_docs/test_build.py

```diff
@@ -0,0 +1,199 @@
+import json
+import logging
+from pathlib import Path
+from textwrap import dedent
+
+import pydantic
+import pytest
+
+from docs.build import (
+    build_schemas_from_source,
+    dump_json_schema,
+    get_schema_from_module,
+    load_schema_module,
+)
+
+
+def test_load_schema_module(tmp_path):
+    pth = Path(tmp_path) / "foo.py"
+    pth.write_text(
+        dedent(
+            """
+FOO = 1
+        """
+        )
+    )
+
+    module = load_schema_module(pth)
+    assert module.FOO == 1
+
+
+def test_get_schema_from_module(tmp_path):
+    pth = Path(tmp_path) / "bar.py"
+    pth.write_text(
+        dedent(
+            """
+import pydantic
+class RequirerSchema(pydantic.BaseModel):
+    foo = 1
+class ProviderSchema(pydantic.BaseModel):
+    foo = 2
+class Foo(pydantic.BaseModel):
+    foo = 3
+class Bar:
+    pass
+"""
+        )
+    )
+    module = load_schema_module(pth)
+
+    requirer_schema = get_schema_from_module(module, "RequirerSchema")
+    assert requirer_schema.__fields__["foo"].default == 1
+    provider_schema = get_schema_from_module(module, "ProviderSchema")
+    assert provider_schema.__fields__["foo"].default == 2
+    foo_schema = get_schema_from_module(module, "Foo")
+    assert foo_schema.__fields__["foo"].default == 3
+
+    # fails because it's not a pydantic model
+    with pytest.raises(TypeError):
+        get_schema_from_module(module, "Bar")
+
+    # fails because it's not found in the module
+    with pytest.raises(NameError):
+        get_schema_from_module(module, "Baz")
```

Comment:
This test does too much, each test should validate only 1 behavior and fail if this behavior is not respected.